ai/src/train.py

```diff
@@ -9,48 +8,74 @@
 
 from algorithms.factory import get_agent
 from algorithms.agent_interface import SpiceAIAgent
-from cleanup import directories_to_delete
 from connector.manager import ConnectorManager
 from data_manager.base_manager import DataManagerBase
 from data_manager.time_series_manager import TimeSeriesDataManager
 from exec import somewhat_safe_eval
-from exception import DataSourceActionInvalidException, LawInvalidException, RewardInvalidException
+from exception import (
+    DataSourceActionInvalidException,
+    LawInvalidException,
+    RewardInvalidException,
+)
 from progress import ProgressBar
 from utils import print_event
 
 
-class Trainer():
+class Trainer:
     TRAINING_LOCK = threading.Lock()
     SAVED_MODELS: Dict[str, Path] = {}
     BASE_URL = "http://localhost:8000/api/v0.1/pods"
 
     def __init__(
-            self, pod_name: str, data_manager: DataManagerBase, connector_manager: ConnectorManager, algorithm: str,
-            number_episodes: int, flight: str, training_goal: str):
+        self,
+        pod_name: str,
+        data_manager: DataManagerBase,
+        connector_manager: ConnectorManager,
+        algorithm: str,
+        number_episodes: int,
+        flight: str,
+        training_goal: str,
+        training_data_dir: str,
+        training_loggers,
+    ):
         self.pod_name = pod_name
         self.data_manager = data_manager
         self.connector_manager = connector_manager
         self.algorithm = algorithm
         self.number_episodes = number_episodes
         self.flight = flight
         self.training_goal = training_goal
+        self.training_data_dir = training_data_dir
+        self.training_loggers = training_loggers
 
         self.action_size = len(data_manager.action_names)
 
-        self.request_url = self.BASE_URL + f"/{pod_name}/training_runs/{flight}/episodes"
+        self.request_url = (
+            self.BASE_URL + f"/{pod_name}/training_runs/{flight}/episodes"
+        )
 
         self.training_episodes = number_episodes
         self.not_learning_threshold = 3
 
+        self.log_dir = str(Path(self.training_data_dir, "log"))
```

Comment:
Letting the log_dir as a Path type would allow to avoid deprecated use of `os.path.mkdir` and `os.path.exist` latter